### PR TITLE
Add ability to switch layouts via joystick buttons

### PR DESCRIPTION
**Pull request by robbi-blechdose**
As discussed in #55, I added this functionality.
Feedback is welcome :grin:

### CODE REVIEW

**Comment by Robbi-Blechdose** [comment]
Sorry to ask again, but is there anything preventing this from being merged?

Or is this project on the backburner? In that case I'd keep to my own fork if that's okay for you.

**Comment by panzi** [comment]
I'm trying to have a look at it this weekend. I'm too tired for anything after work lately.

File: src/button.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@ bool Button::read( QTextStream &stream ) {
 //	at this point, toDefault() has just been called.
 
     //read in a line of text and break it into words
-    QString input = stream.readLine().toLower();
+    QString input = stream.readLine();
```
Comment (panzi) [inline src/button.cpp:22]: Why did you change this here?
And instead of allocating so many temporary strings with every `toLower()` down below why not use `QString::compare(str1, str2, Qt::CaseInsensitive)`?

File: src/button.cpp [hunk 2]
```diff
@@ -51,15 +51,15 @@ bool Button::read( QTextStream &stream ) {
             else return false;
         }
         else if (*it == "layout") {
```
Comment (panzi) [inline src/button.cpp:53]: Why is this not case insensitive now?

File: src/button.cpp (hunk 1 above)
Comment (Robbi-Blechdose) [reply src/button.cpp:22]: The issue here was that setting the entire string to lowercase caused issues with layout handling - since layout names are case sensitive.

Would this be better solved by a separate `inputLower` variable, or should I use the string comparison method you mentioned?
I do apologize for the amount of comparisons, C++ is still a bit new to me so I didn't consider it an issue.

File: src/button.cpp (hunk 2 above)
Comment (Robbi-Blechdose) [reply src/button.cpp:53]: Probably an oversight on my part, I'll fix it together with your other point.

File: src/button.cpp (hunk 1 above)
Comment (panzi) [reply src/button.cpp:22]: Yeah, I thing using `QString::compare(str1, str2, Qt::CaseInsensitive)` would probably be best.

Comment (Robbi-Blechdose) [reply src/button.cpp:22]: Implemented :grin:

File: src/button.cpp (hunk 2 above)
Comment (Robbi-Blechdose) [reply src/button.cpp]: Done.

**Comment by robbi-blechdose** [comment]
Thank you very much! :grinning:
